general-agent-module/src/tests.utils.ts

```diff
@@ -95,4 +96,9 @@ export class TestTransactionEvent extends TransactionEvent {
     this.addresses[address] = true;
     return this;
   }
+
+  public addTrace(trace: Trace): TestTransactionEvent {
+    this.traces.push(trace);
+    return this;
+  }
```

Comment:
I think the idea with these methods is to make tests more readable and smooth to write. If I still need to create the whole Trace object and passing it as an argument that goal is missed I think. I would rather add parameters to the function and create the trace inside.